examples/optical_flow_methods_convergence.py

```diff
@@ -0,0 +1,363 @@
+- Morph the precipitation field using a given motion field (linear or rotor) to
+  generate a sequence of moving precipitation patterns.
+- Using the available optical flow methods, retrieve the motion field from the
+  precipitation time sequence (synthetic precipitation observations).
+
+Let's first load the libraries that we will use.
+"""
+import sys
+from datetime import datetime
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.cm import get_cmap
+from scipy.ndimage import uniform_filter
+
+import pysteps as stp
+from pysteps import motion, io, rcparams
+from pysteps.motion.vet import morph
+from pysteps.visualization import plot_precip_field, quiver
+
+################################################################################
+# Load the reference precipitation data
+# -------------------------------------
+#
+# First, we will import a radar composite from the archive.
+# You need the pysteps-data archive downloaded and the pystepsrc file
+# configured with the data_source paths pointing to data folders.
+
+
+# Selected case
+date = datetime.strptime("201505151630", "%Y%m%d%H%M")
+data_source = rcparams.data_sources["mch"]
+
+###############################################################################
+# Load the data from the archive
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+root_path = data_source["root_path"]
+path_fmt = data_source["path_fmt"]
+fn_pattern = data_source["fn_pattern"]
+fn_ext = data_source["fn_ext"]
+importer_name = data_source["importer"]
+importer_kwargs = data_source["importer_kwargs"]
+
+# Find the input files from the archive
+fns = io.archive.find_by_date(date, root_path, path_fmt, fn_pattern, fn_ext,
+                              timestep=5, num_prev_files=0)
+
+# Read the radar composites
+importer = io.get_method(importer_name, "importer")
+reference_field, quality, metadata = io.read_timeseries(fns, importer,
+                                                        **importer_kwargs)
+
+del quality  # Not used
+
+reference_field = np.squeeze(reference_field)  # Remove time dimension
+
+###############################################################################
+# Preprocess the data
+# ~~~~~~~~~~~~~~~~~~~
+
+# Convert to mm/h
+reference_field, metadata = stp.utils.to_rainrate(reference_field, metadata)
+
+# Mask invalid values
+reference_field = np.ma.masked_invalid(reference_field)
+
+# Log-transform the data [dBR]
+reference_field, metadata = stp.utils.dB_transform(reference_field,
+                                                   metadata,
+                                                   threshold=0.1,
+                                                   zerovalue=-15.0)
+
+print("Precip. pattern shape: " + str(reference_field.shape))
+
+# This suppress nan conversion warnings in plot functions
+reference_field.data[reference_field.mask] = np.nan
+
+plot_precip_field(reference_field, geodata=metadata, title="Reference field")
```

Comment:
This won't work out, since `plot_precip_field`assumes that the input field is in units of mm/h by default. Unfortunately the method doesn't support dBR, so you need to plot the field before the log transformation.

Reply:
Fix in 8b6eeef